src/Type/Php/RangeFunctionReturnTypeExtension.php

```diff
@@ -100,6 +100,12 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 			return new ArrayType(new IntegerType(), new FloatType());
 		}
 
+		if ($startType instanceof StringType
+			&& $endType instanceof StringType
+			&& $stepType instanceof IntegerType) {
+			return new ArrayType(new IntegerType(), new StringType());
```

Comment:
Hi, I'd like the suggested logic from https://github.com/phpstan/phpstan/issues/2378#issuecomment-523173173. These `if` statements won't cover all cases.